### PR TITLE
[PART 1] Implement Metrics Exporter 

### PR DESCRIPTION
**Pull request by lzchen**
Part 1 of the Metrics Exporter implementation

This includes asynchronous polling of metrics on a different thread periodically to be exported to Azure Monitor.

Also includes stats -> metrics migration work. Metrics now uses the data models as defined by https://github.com/open-telemetry/opentelemetry-specification/pull/137
For PART 1, only the "measure" data models are implemented for raw data measurements
As well, aggregation is done via the exporter API, and only COUNT and SUM aggregation types are implemented.

TODO: 
- Implement "metrics" data models for pre-aggregated data
- Implement "Distribution" and "Last Value" aggregations
- Make recording of metrics asynchronous (https://github.com/census-instrumentation/opencensus-python/issues/684)
- Make recording of metrics thread-safe (https://github.com/census-instrumentation/opencensus-python/issues/686)
- Out-of-the-box metrics (such as requests sent)
- Tags

Take a look at the new usage in https://github.com/lzchen/opencensus-python/blob/metrics/contrib/opencensus-ext-azure/examples/metrics/measure.py

### CODE REVIEW

File: contrib/opencensus-ext-azure/README.rst [hunk 1]
```diff
@@ -72,57 +72,40 @@ You can enrich the logs with trace IDs and span IDs by using the `logging integr
...
-                                        [],
-                                        CARROTS_MEASURE,
-                                        aggregation_module.CountAggregation())
-
+        from opencensus.metrics.export import meter
+        from opencensus.metrics.export import measure
+        from opencensus.metrics.export import metrics_producer
+
+        mp = metrics_producer.metrics_producer
+        meter = mp.meter
```
Comment (reyang) [inline contrib/opencensus-ext-azure/README.rst]: Would this shadow the `from opencensus.metrics.export import meter`?

**Comment by c24t** [comment]
Some notes from our offline conversation: I'd recommend splitting this up into one PR to get azure metrics export working and future PRs to make changes to the metrics package. The exporter changes should be self-contained and quick to merge, the metrics API changes will require a lot more work.

File: opencensus/metrics/export/measurement.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from opencensus.ext.azure.common import utils
+
+class Measurement(object):
+    """ A measurement is an object with a value and timestamp attached to it
+        It represents a single raw data point
+        The timestamp is set to the current time that the Measurement is created
+
+    :type value: long or double
+    :param value: value of the measurement
+
```
Comment (c24t) [inline opencensus/metrics/export/measurement.py]: There are some formatting problems here -- too-long lines, extra spaces, leading indentation. Lint seems not to catch these though.

File: opencensus/metrics/export/measurement.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    """
+    def __init__(self, value):
+        self._value = value
+        self._timestamp = utils.to_iso_str()
+
+    @property
+    def value(self):
+        """The value of the current measurement"""
+        return self._value
+
```
Comment (c24t) [inline opencensus/metrics/export/measurement.py]: Some lint errors here and elsewhere, try:

    pylint --include-naming-hint=y $(git diff --name-only master)

File: opencensus/metrics/export/meter.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    """ Meter is a simple, interface that allows users to record measurements (metrics).
+
+        There are two ways to record measurements:
+
+        Record raw measurements, and defer defining the aggregation and the labels for the exported
+               Metric. This should be used in libraries like gRPC to record measurements like
+               "server_latency" or "received_bytes".
+        Record pre-defined aggregation data (or already aggregated data). This should be used to
+              report cpu/memory usage, or simple metrics like "queue_length".
+    """  # noqa
```
Comment (c24t) [inline opencensus/metrics/export/meter.py]: Why not just fix the docstring?

File: opencensus/stats/stats.py [hunk 5]
```diff
@@ -14,12 +14,11 @@
 
 from datetime import datetime
 
-from opencensus.metrics.export.metric_producer import MetricProducer
 from opencensus.stats.stats_recorder import StatsRecorder
 from opencensus.stats.view_manager import ViewManager
 
 
-class _Stats(MetricProducer):
+class _Stats(object):
```
Comment (c24t) [inline opencensus/stats/stats.py]: We still need stats to be a metrics producer until deprecating stats completely.

**Comment by lzchen** [comment]
Closing and splitting work into different parts. Implementing Metrics Exporter first via Stats as discussed.

**Pull request closed without merging**